fix(defaults): Process help and version defaults better

markcarver opened the pull request:
The predefined `help` and `version` options are hardcoded to the lowercase characters when checking if they were provided.

This causes issues if other options like `host` or `verbose` are provided first, which actually does cause the `help` and `version` short options to be uppercased, but are ultimately ignored.

## Summary of changes:

#### Fixes

- Hard coded `help`/`version` short options. If someone provides `-v` for `verbose`, they actually get the version printed out and the process exits (this may also possibly help with #103 somewhat):
  Before: `foo bar -v` -> `1.0.0`
  After: `foo bar -v` -> `Verbose output from bar`
- Options passed before command:
  Before: `foo -dv bar -f file.txt` -> `foo-bar -f file.txt`
  After: `foo -dv bar -f file.txt` -> `foo-bar -dv -f file.txt`
- Bug in `generateDetails` where it was overriding the `usage` property on the items it was generating, which in turn was also altering the original object and causing the tests to bork.
- Updated documentation

#### Added

- An `exit` config option like was described in #95 (needed for tests)

#### Removed

- The `parent` parameter that was added to `checkVersion` in #96. Since this was only introduced in 3.0.2 and has remained an internal method (not exposed until now), there's no real risk of breaking BC.
- The `pkginfo` dependency - This module is unnecessary and, quite frankly, a little too presumptuous in populating metadata onto the module object. While, perhaps useful for other projects, it is overkill for simply attempting to retrieve the main module's version.

ntwcklng commented:
Awesome PR! Would you mind and add tests for both cases like `host` and `verbose`?

markcarver commented:
Ok, I've gone ahead and added separate tests (which really need some love and to be separated BTW).

I ended up having to implement an `exit` config option so we can test the actual flags without the process exiting (which pretty much fixes #95).

Found a nifty little bug in `generateDetails` where it was overriding the `usage` property on the items it was generating, which in turn was also altering the original object and causing the tests to bork.

Removed the `parent` parameter that was added to `checkVersion` in #96. Since this was only introduced in 3.0.2 and has remained an internal method (not exposed until now), there's no real risk of breaking BC.

Updated the docs as well.

markcarver commented:
Seems Travis is having some network issues. Any chance the last commit test can be restarted?

markcarver commented:
Cleaned up the version code a bit so it uses `process.mainModule` instead of `module.parent`.

I also went ahead and provided a summary of changes for the PR.

I think I'm done tinkering with this, for now, don't want to scope creep too much.
